### PR TITLE
Fixed type size, glHint and added more check null

### PR DESCRIPTION
**Pull request by GermanAizek**
@sfan5, review pr changes and feedback me.

### CODE REVIEW

**Comment by GermanAizek** [comment]
@sfan5,
don't forget to do review this PR, thanks.

**Comment by sfan5** [comment]
Which of these changes is a security improvement?

File: source/Irrlicht/CIrrDeviceWin32.cpp [hunk 1]
```diff
@@ -1390,7 +1390,7 @@ void CIrrDeviceWin32::handleSystemMessages()
 
 	while (PeekMessage(&msg, NULL, 0, 0, PM_REMOVE))
 	{
-		if (ExternalWindow && msg.hwnd == HWnd)
+		if (ExternalWindow)
```
Comment (sfan5) [inline source/Irrlicht/CIrrDeviceWin32.cpp:1393]: this is a change in behaviour

**Comment by GermanAizek** [comment]
> Which of these changes is a security improvement?

protection against null reference exception

File: source/Irrlicht/CIrrDeviceWin32.cpp (hunk 1 above)
Comment (GermanAizek) [reply source/Irrlicht/CIrrDeviceWin32.cpp]: fix it

**Review by sfan5 — APPROVED** [review]

**Pull request merged**